### PR TITLE
kernel/binary_manager : Use the structure for load_binary

### PR DESCRIPTION
**Pull request by YiHyunJin**
This commit have included the necessary information in the structure
for the binary load.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [f56a99edb56c7867e1048ef44876c9a70bcb83b1] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f56a99edb56c7867e1048ef44876c9a70bcb83b1] - Code Rule Check Result:
os/include/tinyara/binary_manager.h:108: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/tinyara/binary_manager.h:109: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/tinyara/binary_manager.h:110: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/tinyara/binary_manager.h:111: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/tinyara/binary_manager.h:112: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Target : [0590ed9654431f0919406c67c8355baf59afb4e4] - Code Rule Check Result:
os/include/tinyara/binary_manager.h:108: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/tinyara/binary_manager.h:109: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/tinyara/binary_manager.h:110: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/tinyara/binary_manager.h:111: ERROR: [IDT_M_TAB] please, no space before tabs
os/include/tinyara/binary_manager.h:112: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Target : [0590ed9654431f0919406c67c8355baf59afb4e4] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d29d5f8ada78080ef8fe973728414963013b457b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d29d5f8ada78080ef8fe973728414963013b457b] - Code Rule Check (C++) OK.

File: os/binfmt/binfmt_loadbinary.c [hunk 1]
```diff
@@ -58,34 +57,31 @@ extern void mpu_user_intsram_context(uint32_t region, uintptr_t base, size_t siz
...
- *
- *   offset   - The offset from which ELF binary has to be read in MTD partition.
+ *   filename  - The path to the program to be executed. If
+ *               CONFIG_LIB_ENVPATH is defined in the configuration, then
+ *               this may be a relative path from the current working
+ *               directory. Otherwise, path must be the absolute path to the
+ *               program.
  *
- *   ram_size - The size of RAM partition required by this app
+ *   load_attr - This structure contains the information that the binary
```
Comment (jeongchanKim) [inline os/binfmt/binfmt_loadbinary.c]: I think binary manager needs the load_attr to load, not binary.

File: os/include/tinyara/binary_manager.h [hunk 2]
```diff
@@ -103,6 +103,16 @@ struct binary_info_s {
 };
 typedef struct binary_info_s binary_info_t;
 
+/* The structure of load attr configuration */
+struct load_attr_s {
+	uint32_t size;				/* The size of ELF binary to be loaded */
```
Comment (jeongchanKim) [inline os/include/tinyara/binary_manager.h]: I think elf_size or bin_size is better.

File: os/include/tinyara/binfmt/binfmt.h [hunk 3]
```diff
@@ -371,20 +373,18 @@ int binfmt_exit(FAR struct binary_s *bin);
...
  *
  * Returned Value:
  *   This is an end-user function, so it follows the normal convention:
  *   It returns the PID of the exec'ed module.  On failure, it returns
  *   -1 (ERROR) and sets errno appropriately.
  *
  ****************************************************************************/
-int load_binary(FAR const char *filename, size_t binsize, size_t offset, size_t ram_size, size_t stack_size, uint8_t priority);
+#ifdef CONFIG_BINARY_MANAGER
+int load_binary(FAR const char *filename, FAR struct load_attr_s *load_attr);
```
Comment (jeongchanKim) [inline os/include/tinyara/binfmt/binfmt.h]: You defined `load_attr_t` type and used in binary_manager.h 
How about unifying the type?

File: os/binfmt/binfmt_loadbinary.c (hunk 1 above)
Comment (YiHyunJin) [reply os/binfmt/binfmt_loadbinary.c]: I modified it.

File: os/include/tinyara/binary_manager.h (hunk 2 above)
Comment (YiHyunJin) [reply os/include/tinyara/binary_manager.h]: I modified it.

File: os/include/tinyara/binfmt/binfmt.h (hunk 3 above)
Comment (YiHyunJin) [reply os/include/tinyara/binfmt/binfmt.h]: ok. I modified it.

**Comment by seinfra** [comment]
Target : [42f922881536bfc2296f1731ebd7cf9c9e2949fd] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [42f922881536bfc2296f1731ebd7cf9c9e2949fd] - Code Rule Check OK.

**Pull request merged**